src/core/settings.h

```diff
@@ -382,20 +382,97 @@ enum class GPUAccuracy : u32 {
     Extreme = 2,
 };
 
+extern bool configuring_global;
+
+template <typename Type>
+class Setting {
+public:
+    Setting() {
+        use_global = true;
+    }
+    Setting(Type val) {
+        use_global = true;
+        global = val;
+    }
+    ~Setting() = default;
+    void SetGlobal(bool to_global) {
+        use_global = to_global;
+    };
+    bool UsingGlobal() const {
+        return use_global;
+    };
+    Type GetValue(bool need_global = false) const {
+        if (use_global || need_global) {
+            return global;
+        }
+        return local;
+    };
+    void SetValue(const Type& value) {
+        if (use_global) {
+            global = value;
+        } else {
+            local = value;
+        }
+    };
+    operator Type() const {
+        return GetValue();
+    };
+    Type operator=(const Type& b) {
+        SetValue(b);
+        return b;
+    };
+
+private:
+    bool use_global{};
+    Type global{};
+    Type local{};
+};
+
 struct Values {
+    // Audio
+    std::string audio_device_id;
+    std::string sink_id;
+    bool audio_muted;
+    Setting<bool> enable_audio_stretching;
+    Setting<float> volume;
+
+    // Core
+    Setting<bool> use_multi_core;
+
+    // Renderer
+    Setting<RendererBackend> renderer_backend;
+    Setting<bool> renderer_debug;
+    Setting<int> vulkan_device;
+
+    Setting<u16> resolution_factor = Setting(static_cast<u16>(1));
```

Comment:
No need to cast, use 1U instead
```suggestion
    Setting<u16> resolution_factor = Setting(1U);
```

Reply:
Build failed since it is expecting a short unsigned int, not an unsigned int. Is there a way to write a short unsigned int literal?

Reply:
Using u16(1) would be your best choice then in this case. There's no clean way to represent an unsigned short using an int literal

Reply:
Reverting because that change causes ctest to fail.